### ISSUE TITLE
Unable to push manifest list to quay.io

### ISSUE THREAD

**Issue by w4tsn** [containers/podman#8353]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

I'm trying to create a (multi-arch) manifest with two images and push that to quay.io. The first attempt works without any issues, but if the manifest already exists on the destination, the process fails on quay.io while it does work on docker.io. It could be that the issue therefore lies with quay.io and I've to report it there too.

**Steps to reproduce the issue:**

There is an example on quay.io which you can use to reproduce. It's using the images `quay.io/othermo/kapacitor-container:master-aarch64` and `quay.io/othermo/kapacitor-container:master-amd64`
                                                                                                           
1. `podman pull quay.io/othermo/kapacitor-container:master-amd64 quay.io/othermo/kapacitor-container:master-aarch64`
2. `podman manifest create quay.io/<user>/<repo>:<tag>`
3. `podman manifest add "quay.io/<user>/<repo>:<tag>"  quay.io/othermo/kapacitor-container:master-amd64 quay.io/othermo/kapacitor-container:master-arch64` 
4. `podman manifest push --all "quay.io/<user>/<repo>:<tag>" "docker://quay.io/<user>/<repo>:<tag>"`

**Describe the results you received:**

After the second push attempt (probably with updated images so there is actually a change to push) the command fails:

```
Getting image list signatures
Copying 2 of 2 images in list
[... lot of copying logs ... ]
Storing signatures
Writing manifest list to image destination
Error: error pushing manifest quay.io/othermo/kapacitor-container:master to docker://quay.io/othermo/kapacitor-container:master: Uploading manifest list failed, attempted the following formats: application/vnd.oci.image.index.v1+json(Error uploading manifest master to quay.io/othermo/kapacitor-container: manifest invalid: manifest invalid), application/vnd.docker.distribution.manifest.list.v2+json(Error uploading manifest master to quay.io/othermo/kapacitor-container: manifest invalid: manifest invalid)
```

**Describe the results you expected:**

I'd expect the manifest push to work.

When using the same command against docker.io it works without a problem.

**Additional information you deem important (e.g. issue happens only occasionally):**

This might be a problem in quay.io, so I'll also investigate in that direction.

The two images used in my case are build using different OSes and versions of podman as well as using different manifest versions. Podman manifest inspect outputs this:

```
{
    "schemaVersion": 2,
    "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
    "manifests": [
        {
            "mediaType": "application/vnd.docker.distribution.manifest.v1+prettyjws",
            "size": 11162,
            "digest": "sha256:f3d3e5028d3157aa4541129753e53b92641918f8c1b28423e2a12fd2403856d2",
            "platform": {
                "architecture": "arm64",
                "os": "linux"
            }
        },
        {
            "mediaType": "application/vnd.oci.image.manifest.v1+json",
            "size": 2014,
            "digest": "sha256:98b32a946c18122edb8fef2a3983e1ba60932d796e784a968c1aa0dbb97b7f27",
            "platform": {
                "architecture": "amd64",
                "os": "linux"
            }
        }
    ]
}
```

**Output of `podman version`:**

```
$ podman version
Version:      2.1.1
API Version:  2.0.0
Go Version:   go1.15.2
Built:        Wed Oct  7 18:21:20 2020
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.16.1
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.0.21-3.fc33.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.21, commit: 0f53fb68333bdead5fe4dc5175703e22cf9882ab'
  cpus: 8
  distribution:
    distribution: fedora
    version: "33"
  eventLogger: journald
  hostname: echo-carbon
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.8.18-300.fc33.x86_64
  linkmode: dynamic
  memFree: 1925300224
  memTotal: 16529440768
  ociRuntime:
    name: crun
    package: crun-0.15.1-1.fc33.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 0.15.1
      commit: eb0145e5ad4d8207e84a327248af76663d4e50dd
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  os: linux
  remoteSocket:
    path: /run/user/1000/podman/podman.sock
  rootless: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.1.4-4.dev.giteecccdb.fc33.x86_64
    version: |-
      slirp4netns version 1.1.4+dev
      commit: eecccdb96f587b11d7764556ffacfeaffe4b6e11
      libslirp: 4.3.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.0
  swapFree: 12650012672
  swapTotal: 12650012672
  uptime: 2h 48m 16.35s (Approximately 0.08 days)
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - registry.centos.org
  - docker.io
store:
  configFile: /home/w4tsn/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: fuse-overlayfs-1.2.0-1.fc33.x86_64
      Version: |-
        fusermount3 version: 3.9.3
        fuse-overlayfs: version 1.1.0
        FUSE library version 3.9.3
        using FUSE kernel interface version 7.31
  graphRoot: /home/w4tsn/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 67
  runRoot: /run/user/1000
  volumePath: /home/w4tsn/.local/share/containers/storage/volumes
version:
  APIVersion: 2.0.0
  Built: 1602087680
  BuiltTime: Wed Oct  7 18:21:20 2020
  GitCommit: ""
  GoVersion: go1.15.2
  OsArch: linux/amd64
  Version: 2.1.1
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-2.1.1-12.fc33.x86_64
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide?**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Podman version of the buildah-image running on AWS:

```
$ podman version
Version:      2.0.4
API Version:  1
Go Version:   go1.14.6
Built:        Thu Jan  1 00:00:00 1970
OS/Arch:      linux/amd64

$ podman info --debug
host:
  arch: amd64
  buildahVersion: 1.15.0
  cgroupVersion: v2
  conmon:
    package: conmon-2.0.19-1.fc32.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.19, commit: 5dce9767526ed27f177a8fa3f281889ad509fea7'
  cpus: 8
  distribution:
    distribution: fedora
    version: "32"
  eventLogger: file
  hostname: a5b9d660545f
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 5.8.18-300.fc33.x86_64
  linkmode: dynamic
  memFree: 747347968
  memTotal: 16529440768
  ociRuntime:
    name: crun
    package: crun-0.14.1-1.fc32.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 0.14.1
      commit: 598ea5e192ca12d4f6378217d3ab1415efeddefa
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  os: linux
  remoteSocket:
    path: /run/podman/podman.sock
  rootless: false
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 12650012672
  swapTotal: 12650012672
  uptime: 3h 17m 16.97s (Approximately 0.12 days)
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - registry.centos.org
  - docker.io
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: vfs
  graphOptions: {}
  graphRoot: /var/lib/containers/storage
  graphStatus: {}
  imageStore:
    number: 0
  runRoot: /var/run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 1
  Built: 0
  BuiltTime: Thu Jan  1 00:00:00 1970
  GitCommit: ""
  GoVersion: go1.14.6
  OsArch: linux/amd64
  Version: 2.0.4
```

This issues has similarities in appearance but differs from #8242 in that the initial push (using --all) works, but subsequent pushes fail. Also only the podman command was tried, working against docker.io but not quay.io.

**Comment by mheon** [comment]
@QiWang19 PTAL

@rhatdan This seems like a good candidate for the error clarification work we were talking about, that error is the stuff of nightmares.

**Comment by rhatdan** [comment]
Opened up a PR to cleanup the error a wee bit.

https://github.com/containers/podman/pull/8370

**Comment by QiWang19** [comment]
@w4tsn For updating the image in the manifest list and pushing it. Will the workaround in this [explanation](https://lists.podman.io/archives/list/podman@lists.podman.io/message/EIYJSUW4FM4QXSESJK4GAUA74GOHLCUF/) work for you?

**Comment by w4tsn** [comment]
apparently yes, since I'm the one who asked the question initially :D

So just to have it here for reference what I had to do to get it working:

- add `--format docker` to my build commands
- add `--format v2s2` to my push commands
- (maybe) update buildah from a version from july to latest

So quay.io does not support OCI manifest type, only docker. That's way I had to ensure that my types are set to docker.

So besides the update of the error reporting of podman I don't know if podman could do a lot about this issue so I suppose it can be closed.
